### PR TITLE
Fix int take

### PR DESCRIPTION
**Pull request by siscia**
As mention in #3473

I forked from the branch `v1.0`

Unfortunately I wasn't able to test the change, which should be necessary.

However when I was trying to test I got the following error:

```
simo@simo:~/elixir$ ls
bin              LEGAL    Makefile   rebar.config  VERSION
CHANGELOG.md     lib      README.md  RELEASE.md
CONTRIBUTING.md  LICENSE  rebar      src
simo@simo:~/elixir$ make clean
cd lib/elixir && ../../rebar clean
==> elixir (clean)
rm -rf ebin
rm -rf lib/*/ebin
rm -rf lib/elixir/test/ebin
rm -rf lib/*/tmp
rm -rf lib/mix/test/fixtures/git_repo
rm -rf lib/mix/test/fixtures/deps_on_git_repo
rm -rf lib/mix/test/fixtures/git_rebar
rm -rf lib/elixir/src/elixir.app.src
simo@simo:~/elixir$ make test
==> elixir (compile)
unexpected error compiling src/elixir_parser.yrl
{'EXIT',{undef,[{yecc,file,
                      ["src/elixir_parser.yrl",
                       [{parserfile,"src/elixir_parser.erl"},
                        {report,true},
                        {verbose,false},
                        {return,true}]],
                      []},
                {rebar_erlc_compiler,compile_xrl_yrl,5,
                                     [{file,"src/rebar_erlc_compiler.erl"},
                                      {line,429}]},
                {rebar_base_compiler,compile,3,
                                     [{file,"src/rebar_base_compiler.erl"},
                                      {line,121}]},
                {rebar_base_compiler,compile_worker,3,
                                     [{file,"src/rebar_base_compiler.erl"},
                                      {line,194}]}]}}
ERROR: compile failed while processing /home/simo/elixir/lib/elixir: rebar_abort
make: *** [erlang] Error 1
```

Should I open a new issues or it is a know one ?

### CODE REVIEW

File: lib/elixir/lib/enum.ex [hunk 1]
```diff
@@ -1731,11 +1731,11 @@ defmodule Enum do
     []
   end
 
-  def take(collection, count) when is_list(collection) and count > 0 do
+  def take(collection, count) when is_list(collection) and is_integer(n) and count > 0 do
```
Comment (josevalim) [inline lib/elixir/lib/enum.ex]: The variable here is `count` and not `n`.

File: lib/elixir/lib/enum.ex [hunk 2]
```diff
@@ -1731,11 +1731,11 @@ defmodule Enum do
...
     do_take(collection, count)
   end
 
-  def take(collection, count) when count > 0 do
+  def take(collection, count) when count > 0 and is_integer(n) do
```
Comment (josevalim) [inline lib/elixir/lib/enum.ex]: The variable here is `count` and not `n`.

File: lib/elixir/lib/enum.ex [hunk 3]
```diff
@@ -1747,7 +1747,7 @@ defmodule Enum do
     :lists.reverse(res)
   end
 
-  def take(collection, count) when count < 0 do
+  def take(collection, count) when count < 0 and is_integer(n) do
```
Comment (josevalim) [inline lib/elixir/lib/enum.ex]: The variable here is `count` and not `n`.

**Comment by josevalim** [comment]
The error seems to be related to your erlang installation. If you are using something like debian or ubuntu, please make sure erlang-syntax-tools and erlang-parser-tools or similar things have been installed.

**Comment by lexmag** [comment]
Since it's proposed to `v1.0` it probably should be the cherry-picks of original commits from the master branch, however I believe @josevalim is in charge of doing that:

> I wlll backport in case we decided to do another v1.0 release (which is unlikely).

**Comment by eksperimental** [comment]
@josevalim how come all builds in CI failed for the second commmit https://travis-ci.org/elixir-lang/elixir/builds/70613297

**Comment by josevalim** [comment]
The second commit is invalid. And we can no longer backport the changes, so this PR is welcome. @siscia can you please squash your changes into?

**Comment by siscia** [comment]
@josevalim Sorry, what changes you need ?

As soon as my connection will download the necessary tools to run the unit test I will push again.

**Comment by josevalim** [comment]
@siscia I need you to squash all of your commits into a single one. Currently we have 3 commits in the pull request, it would be awesome if you could squash them into one. Thank you!

**Comment by siscia** [comment]
:smile: 

@josevalim We should be good, tomorrow I will add the test, you want me to squash those too ?

**Comment by josevalim** [comment]
We don't usually test that it won't pass the guard, so this is great as is, thank you.

**Comment by siscia** [comment]
@josevalim,

You are extremely welcome.

However, do not test the guard bring the necessity of this same pull request.

It is obvious that the guard will work, it is not so obvious that the guard will be in place.

Your call :)

**Comment by josevalim** [comment]
Sure, please send another PR then tomorrow. I was just stating what we do in Elixir generally. :)

**Pull request merged**